### ISSUE TITLE
Request: Option to change the shortcut keys

### ISSUE THREAD

**Issue by PunnyBoi** [FreeTubeApp/FreeTube#251]
Is it possible to add an option for changing the shortcut keys? I started using FreeTube just a few days back, and I miss the {shift+.} and {shift+,} keys (as in youtube) to change the playback speed. Also, flashing the playback speed on the screen (when it is altered), would be very helpful.

**Comment by BloodRaven0** [comment]
Big yes to this suggestion! Sad it's been more than 4 years since it's been opened.

**Comment by pseudomonas** [comment]
This would be really helpful - or at least having:
→skip forwards
←skip backwards
↑ volume up
↓ volume down 
[spacebar] pause/play
as youtube does, and Netflix, and VLC, and I think a fair number of other players.

**Comment by HelmicNewciv** [comment]
In particular, having the option to bind navigation in the app to keys would be really useful.  I try to avoid using my mouse too much to mitigate pain and I'm able to find vim-like bindings in applications for most things, but there isn't really one for watching YouTube other than navigating their website through qutebrowser or similar.  It would be nice to, by default, have the arrow keys select videos or options, and have the option to configure vim-ish bindings (namely hjkl, shift-H and shift-L being back and forward in most keyboard-driven browsers and extensions, / to search).

It would also help with making gamepad navigation easier, like on the Steam Deck, as it's a lot easier to bind the left stick to act as arrow keys and pick a video than to have to use the trackpads to click something on a small screen or having to let go of the device with one hand to tap something.

**Comment by nemya9066** [comment]
Same request here, just have a menu and allow the user to choose the keybinding that he wants.
Big thanks to anyone already helping on implementing this.

**Comment by cybergazer** [comment]
Really hope this is added soon. This can be done extremely easily if FreeTube is bundled with the Vimium chrome extension, and would work extremely well because it's the same keybindings as vimium in the browser, so the end user won't have to learn anything new if they learn vimium, and it would take almost 0 extra development time since you just bundle the unpacked extension with it.

**Comment by Walter-o** [comment]
I always use `j` and `k` to browse up and down
`J` and `K` to browse up and down quickly
`m` and `,` to go forwards or backwards.

If vimium was added as a drop-in, i could easliy do this as it already supports custom keybinds.

Going the custom route is probably better in the long run but will cause a ton of issues / bugs that have to be fixed.

**Comment by PietJanHein** [comment]
Hotkeys for actions changing profiles and selecting videos would be a great addition to FreeTube.

**Comment by jasiralavi** [comment]
This would also be like a reference for users to learn what keyboard shortcuts are available.

**Comment by boredboardboared** [comment]
been 6 years since this request. any update?

**Comment by ju-klein** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc 
Yeah, so I think I can work on this with a little guidance from everybody.
If you guys are alright with that, please assign this to me. But I might need a little time, I'm a rookie at Open Source still!

( Edit: @absidue , @ChunkyProgrammer - sorry for tagging you guys as well, but Iḿ unsure what the protocol is here. :( )

Edit2, 11/28/2025: Since I did not receive any feedback AND the issue is still open, I'll break the rules first and ask for forgiveness later:
I'll try to implement this! Edit3: This might take a little time, though. Please bear with me. I _will_ report back in case I cannot do this!

**Comment by absidue** [comment]
@eightmeow Your comment about disabling keyboard shortcuts is already covered by #992 which we have already said we won't implement (as mentioned there if you want to remove keyboard shortcuts, you should modify the code and create a custom build for yourself with that functionality removed). Additionally all of FreeTube's settings are configured through the UI, so we definitely won't be switching to a file based config system just for keyboard shortcuts.

**Comment by eightmeow** [comment]
> [@eightmeow](https://github.com/eightmeow) Your comment about disabling keyboard shortcuts is already covered by [#992]

Because #992 was NOT what I wanted (a way to turn off shortcuts) I followed the link to #251 (allow bindings to be changed), which HASN'T been closed, and posted my comment there, suggesting both an implementation and clarifying that I want to be able to change the binding for SOME (not ALL) keys to nothing. My comment was marked off-topic, and when I asked why, it was marked duplicate.

Whatever. Nevermind.

> Additionally all of FreeTube's settings are configured through the UI, so we definitely won't be switching to a file based config system just for keyboard shortcuts.

This is a response that would have satisfied me. I thought a UI solution may be more difficult, but if that's the approach, that's fine.

**Comment by ju-klein** [comment]
Happy new year everybody,
s9omething else has come up (which takes precedence over open source work), so I will not be able to work on this until at least mid-year. If anyone else wants to snatch this, feel free! Sorry about this!

**Comment by samitsingh0411** [comment]
I don't understand why this isn't a thing yet, having custom shortcuts should be like the most basic feature that the app should have. They have achieved much harder stuff this doesn't exists yet. I use s and d to speed up and slow down video in youtube, can't do that here, gotta use their version 😭.

It would be very nice too have this, right now this application is amazing but lack of shortcut makes it feel less professional. Even through it's probably one of the best ones out there for it's usecase.
